main/src/ca/uwaterloo/flix/language/ast/Ast.scala

```diff
@@ -532,4 +532,44 @@ object Ast {
 
   }
 
+  /**
+    * Represents the text of a variable.
+    */
+  sealed trait VarText {
+
+    /**
+      * A measure of precision of the text.
+      */
+    private def precision: Int = this match {
+      case VarText.Absent => 0
+      case VarText.Synthetic(_) => 1
+      case VarText.Text(_) => 2
+    }
```

Comment:
Named | FakeName | Unnamed?

Reply:
SourceText | ErsatsText

Reply:
Label = Text | Hint

Reply:
I don't like "label" since it's too easy to ignore.

SourceText | FallbackText ?

Reply:
Text = Absent | SourceText | FallbackText

And then we want to remove Absent in the next PR? Seems reasonable to me.